taqueria-vscode-extension-web-ui/src/AllEditors.tsx

```diff
@@ -0,0 +1,369 @@
+			],
+			'args': [
+				{
+					'prim': 'string',
+				},
+			],
+		},
+		value: [
+			{
+				string: 'Hello',
+			},
+			{
+				string: 'World',
+			},
+		],
+	},
+	{
+		type: {
+			'prim': 'map',
+			'annots': [
+				'%mapStringInt',
+			],
+			'args': [
+				{
+					'prim': 'string',
+				},
+				{
+					'prim': 'int',
+				},
+			],
+		},
+		value: [
+			{
+				'prim': 'Elt',
+				args: [
+					{
+						'string': 'Hello',
+					},
+					{
+						'int': '42',
+					},
+				],
+			},
+			{
+				'prim': 'Elt',
+				args: [
+					{
+						'string': 'World',
+					},
+					{
+						'int': '43',
+					},
+				],
+			},
+		],
+	},
+	{
+		type: {
+			'prim': 'option',
+			'annots': [
+				'%optionOfInt',
+			],
+			'args': [
+				{
+					'prim': 'int',
+				},
+			],
+		},
+		value: {
+			'prim': 'Some',
+			args: [{
+				'int': '34',
+			}],
+		},
+	},
+	{
+		type: {
+			'prim': 'pair',
+			'annots': [
+				'%pairOfIntString',
+			],
+			'args': [
+				{
+					'prim': 'int',
+				},
+				{
+					'prim': 'string',
+				},
+			],
+		},
+		value: {
+			'prim': 'Pair',
+			'args': [
+				{
+					'int': '42',
+				},
+				{
+					'string': 'Hello',
+				},
+			],
+		},
+	},
+	{
+		type: {
+			'prim': 'pair',
+			'annots': [
+				'%pairOfIntStringStringInt',
+			],
+			'args': [
+				{
+					'prim': 'pair',
+					'args': [
+						{
+							'prim': 'int',
+						},
+						{
+							'prim': 'string',
+						},
+						{
+							'prim': 'string',
+						},
+						{
+							'prim': 'int',
+						},
+					],
+				},
+			],
+		},
+		value: {
+			'prim': 'Pair',
+			'args': [
+				{
+					'prim': 'Pair',
+					'args': [
+						{
+							'int': '42',
+						},
+						{
+							'string': 'Hello',
+						},
+						{
+							'string': 'World',
+						},
+						{
+							'int': '43',
+						},
+					],
+				},
+			],
+		},
+	},
+	{
+		type: {
+			'prim': 'pair',
+			'annots': [
+				'%pairOfListStringListOptionInt',
+			],
+			'args': [
+				{
+					'prim': 'list',
+					'args': [
+						{
+							'prim': 'string',
+						},
+					],
+				},
+				{
+					'prim': 'list',
+					'args': [
+						{
+							'prim': 'option',
+							'args': [
+								{
+									'prim': 'int',
+								},
+							],
+						},
+					],
+				},
+			],
+		},
+		value: {
+			'prim': 'Pair',
+			'args': [
+				[
+					{
+						'string': 'Hello',
+					},
+					{
+						'string': 'World',
+					},
+				],
+				[
+					{
+						'prim': 'None',
+					},
+					{
+						'prim': 'Some',
+						'args': [{
+							'int': '12',
+						}],
+					},
+				],
+			],
+		},
+	},
+	{
+		type: {
+			'prim': 'set',
+			'annots': [
+				'%setOfString',
+			],
+			'args': [
+				{
+					'prim': 'string',
+				},
+			],
+		},
+		value: [
+			{
+				'string': 'Hello',
+			},
+			{
+				'string': 'World',
+			},
+		],
+	},
+];
+
+export const AllEditors = () => {
+	usePageTitle('Test Editors for many Data Types');
+
+	const handleMessage: MichelineEditorMessageHandler = data => {
+		console.log(data);
+	};
```

Comment:
Good practice for production usage: `useCallback`

Reply:
Oh. Good to know. Thanks